### DISCUSSION TITLE
MeshGraphormer Error Problem - No module named 'custom_mesh_graphormer'

### DISCUSSION THREAD

**Discussion by DonDanDojo** [Fannovel16/comfyui_controlnet_aux#377] [Q&A] [answered]
Hello! I have problem with custom mesh graphormer. I've tried reinstalling the nodes but it doesn't help. Could anyone help please?
![3](https://github.com/Fannovel16/comfyui_controlnet_aux/assets/117159388/850033fa-9274-438f-9b66-852f29ef246f)

**Comment by ghost** [comment]
I have similar problem but with `No module named 'controlnet_aux.depth_anything'` and solved it by 
appending the dir path to `PYTHONPATH`, in my case `/mnt/vol1/ComfyUI/custom_nodes/comfyui_controlnet_aux/src/`

**Reply by Nathiboy51** [reply]
Same here 
![image](https://github.com/user-attachments/assets/6cdb367c-5e3a-48cc-9f23-599b16410963)

I tried your fix but I'm not familiar with the process. Could you be more specific @shamikct ?

Of course, I do have yacs installed and all other modules from requirements.txt on the whole PC (no virtual environments)

**Reply by ghost** [reply]
am running ComfyUI on linux and the following worked for me. 
i did the following for making sure that all the `controlnet_aux` modules are accessible:
```sh
export PYTHONPATH=$PYTHONPATH:/mnt/vol1/ComfyUI/custom_nodes/comfyui_controlnet_aux/src/
```
you will have to replace this with the path of your ComfyUI installation and use either the set command on cmd or env command in powershell. 

cmd command 
```sh
set PYTHONPATH=%PYTHONPATH%;C:\mnt\vol1\ComfyUI\custom_nodes\comfyui_controlnet_aux\src\

```
powershell command 
```sh
$env:PYTHONPATH += ";C:\mnt\vol1\ComfyUI\custom_nodes\comfyui_controlnet_aux\src\"
```

if neither of the commands work for you i don't know what else can be done. 


unsolicited advice: use venv instead of installing packages system wide.

**Reply by Nathiboy51** [reply, marked as answer]
Thank you. 
I'm on Windows and realized that _python_embeded_ dir used for Comfy, which is why my first attempts were unsuccessful.
Adding this dir to the PATH didn't fix the _no module_ error.
Now I've used the manual _install.bat_, (which didn't work at first since a previous repo had spaces in its name) so all the requirements are correctly installed.

Unfortunately, I now have another error :

![image](https://github.com/user-attachments/assets/68ad7ed9-4d88-4ccd-91a8-1d83ab5bf3ce)

Edit: I fixed it by cloning _https://huggingface.co/hr16/ControlNet-HandRefiner-pruned/tree/main_ in _\ComfyUI\custom_nodes\comfyui_controlnet_aux\ckpts\hr16\ControlNet-HandRefiner-prune_

Now all is working as expected

**Reply by ghost** [reply]
Just to recap, the _install.bat_ fixed the no module found error and then providing the relevant files to the relevant dir fixed the other error ?

**Reply by Nathiboy51** [reply]
Yes, exactly.

**Reply by DonDanDojo** [reply]
> I have similar problem but with `No module named 'controlnet_aux.depth_anything'` and solved it by appending the dir path to `PYTHONPATH`, in my case `/mnt/vol1/ComfyUI/custom_nodes/comfyui_controlnet_aux/src/`

I probably have the similar problem now.

[ERROR] An error occurred while retrieving information for the 'DepthAnythingPreprocessor' node.
Traceback (most recent call last):
  File "X:\SD Launcher\SD nowy\Packages\ComfyUI\server.py", line 447, in get_object_info
    out[x] = node_info(x)
  File "X:\SD Launcher\SD nowy\Packages\ComfyUI\server.py", line 424, in node_info
    info['input'] = obj_class.INPUT_TYPES()
  File "X:\SD Launcher\SD nowy\Packages\ComfyUI\custom_nodes\comfyui_controlnet_aux\node_wrappers\depth_anything.py", line 8, in INPUT_TYPES
    ckpt_name=INPUT.COMBO(
TypeError: INPUT.COMBO() got an unexpected keyword argument 'resolution'

Could you parahrase what could a noob like me do? :D

**Reply by ghost** [reply]
This error message isn't similar to `no module named`, so can't really help much here.

**Reply by DonDanDojo** [reply]
> This error message isn't similar to `no module named`, so can't really help much here.

Nevermind. It was a typo from its creator. Thanks for reply
